src/Raven.Server/Documents/Replication/IncomingReplicationHandler.cs

```diff
@@ -604,20 +618,37 @@ private void ReceiveSingleDocumentsBatch(DocumentsOperationContext documentsCont
         private void ValidateIncomingReplicationItemsPaths(DataForReplicationCommand dataForReplicationCommand)
         {
             HashSet<Slice> expectedAttachmentStreams = null;
+            
             foreach (var item in dataForReplicationCommand.ReplicatedItems)
             {
-                if (_allowedPathsValidator.ShouldAllow(item) == false)
+                if (_allowedPathsValidator != null)
                 {
-                    throw new InvalidOperationException("Attempted to replicate " + _allowedPathsValidator.GetItemInformation(item) +
-                                                        ", which is not allowed, according to the allowed paths policy. Replication aborted");
+                    if (_allowedPathsValidator.ShouldAllow(item) == false)
+                    {
+                        throw new InvalidOperationException("Attempted to replicate " + _allowedPathsValidator.GetItemInformation(item) +
+                                                            ", which is not allowed, according to the allowed paths policy. Replication aborted");
+                    }
+
+                    switch (item)
+                    {
+                        case AttachmentReplicationItem a:
+                            expectedAttachmentStreams ??= new HashSet<Slice>(SliceComparer.Instance);
+                            expectedAttachmentStreams.Add(a.Key);
+                            break;
+                    }
                 }
 
-                switch (item)
+                if (_preventIncomingSinkDeletions)
                 {
-                    case AttachmentReplicationItem a:
-                        expectedAttachmentStreams ??= new HashSet<Slice>(SliceComparer.Instance);
-                        expectedAttachmentStreams.Add(a.Key);
-                        break;
+                    switch (item)
+                    {
+                        case DocumentReplicationItem { Type: ReplicationBatchItem.ReplicationItemType.DeletedTimeSeriesRange }:
```

Comment:
1. Do we handle deleted revisions? replicating deleted revision might delete the document (not the same as revision tombstones)
2. Time-Series is tricky... we replicate both `DeletedTimeSeriesRange` and segments, segment might contain deleted values. 

cc @ayende thought about TS?

Reply:
We should skip deleting revisions and time series ranges deletes directly.
However, modification to the data (including removing values) is okay.
The user is allowed to mutate the documents. 

Note that this isn't really  a security feature, it is meant to make it easier to manage, not to protect against malicious.

Reply:
Done